summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -196,9 +216,31 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
       StormStatProvider.registerMetrics(jobID, context, countersForSpout)
       SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
     })
-    val stormSpout = tormentaSpout.openHook(registerAllMetrics.get).getSpout
-    val parallelism = getOrElse(stormDag, node, parOpt.getOrElse(DEFAULT_SOURCE_PARALLELISM)).parHint
-    topologyBuilder.setSpout(nodeName, stormSpout, parallelism)
+    val hookedTormentaSpout = tormentaSpout.openHook(registerAllMetrics.get)
+
+    val summerOpt: Option[SummerNode[Storm]] = stormDag.dependantsOf(node.asInstanceOf[StormNode]).collect { case s: SummerNode[Storm] => s }.headOption
+
+    val stormSpout = summerOpt match {
+      //As the spout is being followed by a summer, we do not have map-side aggregation handled in the spout. This means Summer is the place where the aggregation should happen.
+      case Some(s) => {
+        val summerParalellism = getOrElse(stormDag, s, DEFAULT_SUMMER_PARALLELISM)
+        val summerBatchMultiplier = getOrElse(stormDag, s, DEFAULT_SUMMER_BATCH_MULTIPLIER)
+        val keyValueShards = executor.KeyValueShards(summerParalellism.parHint * summerBatchMultiplier.get)
+
+        val summerProducer = s.members.collect { case s: Summer[_, _, _] => s }.head.asInstanceOf[Summer[Storm, _, _]]
```

Comment:
+1 for a comment to explain this. The next person to change this code may not be you and we will forget this reasoning.